packages/navie/src/services/next-step-classification-service.ts

```diff
@@ -0,0 +1,89 @@
+import { z } from 'zod';
+import Message from '../message';
+import { ChatHistory } from '../navie';
+import CompletionService, { ModelType } from './completion-service';
+
+const scoreSchema = z.object({
+  evaluation: z.string(),
+  score: z.number(),
+});
+const NextStepResponseFormat = z.object({
+  nextSteps: z.array(
+    z.object({
+      command: z.enum(['generate', 'diagram', 'plan', 'test', 'explain', 'help']),
+      prompt: z.string(),
+      label: z.string(),
+      reasoning: z.object({
+        relevance: scoreSchema,
+        information: scoreSchema,
+        uniqueness: scoreSchema,
+      }),
+      overallScore: z.number(),
+    })
+  ),
+});
+
+export type NextStepResponse = z.infer<typeof NextStepResponseFormat>['nextSteps'];
+export type NextStep = Omit<NextStepResponse[number], 'reasoning'>;
+
+const SYSTEM_PROMPT = `Your job is to predict the next step(s) for the user given a transcript of their conversation. The transcript will contain a single question and answer, already presented to the user. The users next action will typically be in response to the assistant's answer.
+
+## Commands
+These are the actions that the user may request next:
+- \`generate\`: This will generate code for the user.
+- \`diagram\`: This will generate a (sequence, flow, class, entity relation) diagram for the user. This is useful for visualizing complex concepts, relationships, or processes.
+- \`plan\`: This is useful to create step by step plans to implement a code change in the user's application.
+- \`test\`: This action will generate test cases for the user.
+- \`explain\`: This action will explain parts of the codebase or application behavior.
+- \`help\`: This action will teach the user how to use AppMap Navie.
```

Comment:
A couple of suggestions on this:

1) Put the commands in the general order that the user will use them; for example, plan before generate.
2) In the prompt, propose a few things to the LLM that are fairly concrete. For example, "@generate will often follow @plan". "@explain will often be followed by @diagram". "@test will often follow @generate". Etc.